### DISCUSSION TITLE
[V5] flashMessages return null if validator catch many errors

### DISCUSSION THREAD

**Discussion by PCC94** [adonisjs/core#2488] [Help]
Hi!

I have an issue with flashMessages when a form validation fails, in particular when more than one rules are triggered . 

In this project I have many forms, and some of them have many input with validation rules associated. If I submit the form with only one "invalid" input, validation rules are correctly triggered and flashMessages.all() returns the error message and all the values i inserted in the form.

flashMessages.all() output :
```
{
  "customerId": '398',
  "customerFullname": 'BOLOGNA',
  "estate": 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',< ------- maxlength rule violated
  "building": 'lorem ipsum',
  "service": 'lorem ipsum',
  "floor": 'lorem ipsum',
  "stationing": 'lorem ipsum',
  "operator": 'lorem ipsum',
  "subject": 'lorem ipsum',
  "startAt": '2021-04-15',
  "endAt": '2021-04-15',
  "activityId": '1',
  "errors": {
    "estate": [
     'Too long text input (max 50 characters) '
    ]
  }
}  
```

But, if i try to trigger more than one validation error, for example:

```
 "estate": 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',< ------- maxlength rule violated
  "building": 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',< ------- maxlength rule violated
```

Validation errors are correctly triggered once more. If i log ``error.messages`` i have:

```
{
  estate: [ 'Too long text input (max 50 characters)' ],
  building: ['Too long text input (max 50 characters)']
}
```

... but everytime I inspect ``flashMessages.all()`` in view, it return ``null``. :(

To resolve the problem, i tried with a custom ExceptionHandler, but the results are the same:

```
  public async handle (error, ctx) {
    if (error.code === 'E_VALIDATION_FAILURE') {
      const input = ctx.request.all()
      input.errors = error.messages
      ctx.session.flash(input)
      console.log(ctx.session.flashMessages.all())  // returns null if error.messages has more than one key
      return ctx.response.redirect().back()
    }
  }
```
It seems ctx.session.flash() is not working, but why?

p.s. The session drive is 'cookie' and it appears to work if you use session drive 'file', but i'd prefer to know how to make it work with the cookie session drive. Any suggestions?

**Comment by thetutlage** [comment]
Can you please share a repo with the required code to help us reproduce the issue?

**Comment by leocabeza** [comment]
Hey @PCC94 , have you figured this out? I just had this, but I couldn't solve it

**Comment by McSneaky** [comment]
Most likely, because max cookie size is 4096bytes, which includes all cookie settings for all cookies combined: https://chromestatus.com/feature/4946713618939904

It's a browser implementation thing, can't do anything about it :|
